rosidl_typesupport_opendds_c/resource/msg__type_support_c.cpp.em

```diff
@@ -471,15 +463,10 @@ _@(message.structure.namespaced_type.name)__to_cdr_stream(
     cdr_stream->allocator.deallocate(cdr_stream->buffer, cdr_stream->allocator.state);
     cdr_stream->buffer = static_cast<uint8_t *>(cdr_stream->allocator.allocate(cdr_stream->buffer_length, cdr_stream->allocator.state));
   }
-  // call the function again and fill the buffer this time
-  unsigned int buffer_length_uint = static_cast<unsigned int>(cdr_stream->buffer_length);
-  if (@(__dds_cpp_msg_type_prefix)_Plugin_serialize_to_cdr_buffer(
-      reinterpret_cast<char *>(cdr_stream->buffer),
-      &buffer_length_uint,
-      &dds_message) != true)
-  {
-    return false;
-  }
+  OpenDDS::DCPS::Message_Block_Ptr b(new ACE_Message_Block(size));
+  OpenDDS::DCPS::Serializer serializer(b.get(), false);
+  serializer << dds_message;
```

Comment:
This returns `false` if there was an error and should be checked. Also for the serializer, I think always setting `swap_bytes` to `false` would be a problem for systems with different endiannesses.

Reply:
Also I hope that is the correct plural of "endianness".

Reply:
"... with different byte orders" would be one way to phrase it, but I don't think that's wrong.

To the main point, Mike did some investigation with FastRTPS and it seems like the issue is present there as well.  Since that is the default RMW of ROS2, it is in a way a spec (given that there is no spec for this stuff, at least none that we can find).